### PR TITLE
Assign driver to bus

### PR DESCRIPTION
**Pull request by kelliaUmuhire**
#### What does this PR do?
* Adds endpoint for assigning driver to bus
#### Description of Task to be completed?
* An operator should be able to assign a driver to a bus
#### How should this be manually tested?
* Clone the repo
* Run npm install to install modules
* Run npm run dev to start the development server and the database
* Go to the endpoint `"/api/buses/assign"`
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker/Trello stories?
* [Link](https://trello.com/c/JSMuPtng)

### CODE REVIEW

**Review by fistonfle — APPROVED** [review]
Well implemented

File: src/db/config.ts [hunk 1]
```diff
@@ -3,30 +3,42 @@ import dotenv from "dotenv"
...
-          database: String(process.env.DB_NAME),
-          username: String(process.env.DB_USER),
-          password: String(process.env.DB_PASSWORD),
-          logging: false,
-          sync: {
-              force: false,
-              alter: { drop: false },
-          },
-      })
+// const sequelize =
```
Comment (degide) [inline src/db/config.ts]: Please uncomment these changes to enable the automatic switch between dev and test environmenrs

File: src/db/config.ts [hunk 2]
```diff
@@ -3,30 +3,42 @@ import dotenv from "dotenv"
...
+//         username: String(process.env.DB_USER),
+//         password: String(process.env.DB_PASSWORD),
+//         logging: false,
+//         sync: {
+//           force: false,
+//           alter: { drop: false },
+//         },
+//       });
+
+const sequelize = new Sequelize(
```
Comment (degide) [inline src/db/config.ts]: This can be removed in the favor of the above

File: documentation/buses.yml [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+                allOf:
+                  - $ref: "#/components/schemas/ResBody"
+                  - type: object
+                    properties:
+                      data:
+                        type: object
+                        properties:
+                          count:
+                            type: number
+  /buses/assign:
```
Comment (degide) [inline documentation/buses.yml]: `/buses/asign`. Try to make it meaningful. Try  something like: `/buses/assign-bus-driver`

File: documentation/buses.yml [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+                allOf:
+                  - $ref: "#/components/schemas/ResBody"
+                  - type: object
+                    properties:
+                      data:
+                        type: object
+                        properties:
+                          count:
+                            type: number
+  /buses/drivers:
```
Comment (degide) [inline documentation/buses.yml]: /buses/drivers? Try to make it meaningful. Try something like: /buses/buses-drivers

File: src/routes/buses.router.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+router.put("/:busId", verifyToken, busValidation, updateBusById);
+
+router.get("/bus/:busId", verifyToken, getBusById);
+
+router.get("", verifyToken, getAllBuses);
+
+router.delete("/:busId", verifyToken, deleteBusById);
+
+router.post("/assign", verifyToken, assignValidation, assignDriverToBus);
```
Comment (degide) [inline src/routes/buses.router.ts]: Update this route to meet changes requested in docs

File: src/routes/buses.router.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+router.get("/drivers", verifyToken, getBusDrivers);
```
Comment (degide) [inline src/routes/buses.router.ts]: Update this route to meet changes requested in docs

**Review by degide — CHANGES_REQUESTED** [review]
Well done. Please update according to the requested changes

**Comment by degide** [comment]
Also resolve the conflicts

File: src/db/config.ts (hunk 1 above)
Comment (kelliaUmuhire) [reply src/db/config.ts]: Sure, sure!

File: documentation/buses.yml (hunk 3 above)
Comment (kelliaUmuhire) [reply documentation/buses.yml]: Or maybe `/buses/assign-driver` ?

**Review by degide — APPROVED** [review]

**Review by freddykaberuka — APPROVED** [review]

**Pull request merged**